File: 2024/day8/python/test_problem.py
import unittest
from problem import problem_part1, problem_part2

class TestProblem(unittest.TestCase):
    def test_problem_part1_test1(self):
        input_list = [
            "............",
            "........0...",
            ".....0......",
            ".......0....",
            "....0.......",
            "......A.....",
            "............",
            "............",
            "........A...",
            ".........A..",
            "............",
            "............"
        ]
        result = problem_part1(input_list)
        self.assertEqual(result, 14)


    def test_problem_part2_test1(self):
        input_list = [
            "T.........",
            "...T......",
            ".T........",
            "..........",
            "..........",
            "..........",
            "..........",
            "..........",
            "..........",
            ".........."
        ]
        result = problem_part2(input_list)
        self.assertEqual(result, 9)


    def test_problem_part2_test2(self):
        input_list = [
            "............",
            "........0...",
            ".....0......",
            ".......0....",
            "....0.......",
            "......A.....",
            "............",
            "............",
            "........A...",
            ".........A..",
            "............",
            "............"
        ]
        result = problem_part2(input_list)
        self.assertEqual(result, 34)

File: 2025/day5/julia/TestProblem.jl
using Test

include("Problem.jl")

@testset "Problem" begin
    input_vector = [
        "3-5",
        "10-14",
        "16-20",
        "12-18",
        "",
        "1",
        "5",
        "8",
        "11",
        "17",
        "32"
    ]
    @test problem_part1(input_vector) == 3

    @test problem_part2(input_vector) == 14

    input_vector2 = [
        "2-21",
        "3-5",
        "10-14",
        "16-20",
        "12-18",
    ]
    @test problem_part2(input_vector2) == 20

end